### PR TITLE
Task-41285: Implement drawer for users advanced filter in userManagement

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties [hunk 1]
```diff
@@ -334,7 +334,11 @@ UsersManagement.lastName=Last name
 UsersManagement.email=Email
 UsersManagement.lastConnexion=Last connection
 UsersManagement.lastConnexion.neverConnected=Never connected
+UsersManagement.lastConnexion.connected=Connected
```
Comment (plamarque) [inline webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties]: « connection » in english. not connexion

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java [hunk 2]
```diff
@@ -894,11 +894,11 @@ public int countSpaceMemberIdentitiesByProfileFilter(Space space,
                                       char firstCharacter,
                                       String sortField,
                                       String sortDirection,
-                                      boolean excludeExternal,
                                       boolean IsEnabled,
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java]:
```suggestion
                                      boolean isEnabled,
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java [hunk 3]
```diff
@@ -894,11 +894,11 @@ public int countSpaceMemberIdentitiesByProfileFilter(Space space,
...
+                                      String userType,
                                       long offset,
                                       long limit) {
-    List<String> usernames = getIdentityDAO().getAllIdsByProviderSorted(providerId, firstCharacterFieldName, firstCharacter, sortField, sortDirection, excludeExternal, IsEnabled, offset, limit);
+    List<String> usernames = getIdentityDAO().getAllIdsByProviderSorted(providerId, firstCharacterFieldName, firstCharacter, sortField, sortDirection, IsEnabled, userType, offset, limit);
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java]:
```suggestion
    List<String> usernames = getIdentityDAO().getAllIdsByProviderSorted(providerId, firstCharacterFieldName, firstCharacter, sortField, sortDirection, isEnabled, userType, offset, limit);
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java [hunk 4]
```diff
@@ -141,14 +142,14 @@ public long countIdentityByProvider(String providerId) {
   }
 
   @Override
-  public List<String> getAllIdsByProviderSorted(String providerId, String firstCharacterFieldName, char firstCharacter, String sortField, String sortDirection, boolean excludeExternal, boolean IsEnabled, long offset, long limit) {
-    Query query = getIdentitiesQuerySortedByField(providerId, firstCharacterFieldName, firstCharacter, sortField, sortDirection, excludeExternal, IsEnabled);
+  public List<String> getAllIdsByProviderSorted(String providerId, String firstCharacterFieldName, char firstCharacter, String sortField, String sortDirection, boolean IsEnabled, String userType, long offset, long limit) {
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java]:
```suggestion
  public List<String> getAllIdsByProviderSorted(String providerId, String firstCharacterFieldName, char firstCharacter, String sortField, String sortDirection, boolean isEnabled, String userType, long offset, long limit) {
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java [hunk 5]
```diff
@@ -141,14 +142,14 @@ public long countIdentityByProvider(String providerId) {
...
+    Query query = getIdentitiesQuerySortedByField(providerId, firstCharacterFieldName, firstCharacter, sortField, sortDirection, IsEnabled, userType);
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java]:
```suggestion
    Query query = getIdentitiesQuerySortedByField(providerId, firstCharacterFieldName, firstCharacter, sortField, sortDirection, isEnabled, userType);
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java [hunk 6]
```diff
@@ -141,14 +142,14 @@ public long countIdentityByProvider(String providerId) {
...
     return getResultsFromQuery(query, 0, offset, limit, String.class);
   }
 
   @Override
-  public int getAllIdsCountByProvider(String providerId, boolean excludeExternal, boolean IsEnabled) {
-    Query query = getIdentitiesQueryCount(providerId, excludeExternal, IsEnabled);
+  public int getAllIdsCountByProvider(String providerId, String userType, boolean IsEnabled) {
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java]:
```suggestion
  public int getAllIdsCountByProvider(String providerId, String userType, boolean isEnabled) {
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java [hunk 7]
```diff
@@ -141,14 +142,14 @@ public long countIdentityByProvider(String providerId) {
...
+    Query query = getIdentitiesQueryCount(providerId, userType, IsEnabled);
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java]:
```suggestion
    Query query = getIdentitiesQueryCount(providerId, userType, isEnabled);
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java [hunk 8]
```diff
@@ -364,10 +369,10 @@ private Query getIdentitiesQuerySortedByField(String providerId,
                                                 char firstCharacter,
                                                 String sortField,
                                                 String sortDirection,
-                                                boolean excludeExternal,
-                                                boolean IsEnabled) {
+                                                boolean IsEnabled,
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java]:
```suggestion
                                                boolean isEnabled,
```

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java [hunk 9]
```diff
@@ -339,14 +340,18 @@ public int getSize() throws Exception {
     }
   }
 
-  private Query getIdentitiesQueryCount(String providerId, boolean excludeExternal, boolean IsEnabled) {
+  private Query getIdentitiesQueryCount(String providerId, String userType, boolean IsEnabled) {
```
Comment (rdenarie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/dao/jpa/IdentityDAOImpl.java]:
```suggestion
  private Query getIdentitiesQueryCount(String providerId, String userType, boolean isEnabled) {
```

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 10]
```diff
@@ -274,7 +276,7 @@ public Response getUsers(@Context UriInfo uriInfo,
       return Response.status(HTTPStatus.UNAUTHORIZED).build();
     }
 
-    if (!userACL.getSuperUser().equals(userId) && !RestUtils.isMemberOfAdminGroup() && !excludeExternal) {
+    if (!userACL.getSuperUser().equals(userId) && !RestUtils.isMemberOfAdminGroup() && !userType.equals(INTERNAL)) {
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]:
```suggestion
    if (!userACL.getSuperUser().equals(userId) && !RestUtils.isMemberOfAdminGroup() && userType!=null && !userType.equals(INTERNAL)) {
```

File: component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java [hunk 11]
```diff
@@ -256,8 +258,8 @@ public void stop() {
   public Response getUsers(@Context UriInfo uriInfo,
                            @ApiParam(value = "User name information to filter, ex: user name, last name, first name or full name", required = false) @QueryParam("q") String q,
                            @ApiParam(value = "User status to filter online users, ex: online", required = false) @QueryParam("status") String status,
+                           @ApiParam(value = "User type to filter, ex: internal, external", required = false) @QueryParam("userType") String userType,
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java]: add a defualt value for userType ("internal")

File: webapp/portlet/src/main/webapp/people-list/components/PeopleCardList.vue [hunk 12]
```diff
@@ -165,7 +166,7 @@ export default {
           || this.filter === 'redactor') {
         searchUsersFunction = this.$spaceService.getSpaceMembers(this.keyword, this.offset, this.limitToFetch + 1, this.fieldsToRetrieve, this.filter, this.spaceId);
       } else {
-        searchUsersFunction = this.$userService.getUsers(this.keyword, this.offset, this.limitToFetch + 1, this.fieldsToRetrieve);
+        searchUsersFunction = this.$userService.getUsers(this.keyword, this.offset, this.limitToFetch + 1, this.fieldsToRetrieve, this.userType);
```
Comment (rdenarie) [inline webapp/portlet/src/main/webapp/people-list/components/PeopleCardList.vue]: remove this modification

**Review by rdenarie — CHANGES_REQUESTED** [review]
Add a unit test to UserRestResroucesV1.getUsers : be sure that if you call without userType, you get the list of internal users without error

**Review by rdenarie — APPROVED** [review]

**Pull request merged**